### PR TITLE
doc: Installation guide for Talos

### PR DESCRIPTION
**Pull request by PhilipSchmid**
* Added Helm installation guide for Talos Linux >= 1.5.0
* Listed Talos Linux >=1.5.0 as supported OS

```release-note
doc: Installation guide for Talos
```

### CODE REVIEW

File: Documentation/installation/k8s-install-talos-linux.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  - Cilium's Talos Linux support is only tested with Talos versions ``>=1.5.0``.
+  - As Talos `does not allow loading Kernel modules`_ by Kubernetes workloads, ``SYS_MODULE`` needs to be dropped from the Cilium default capability list.
+
+.. _`does not allow loading Kernel modules`: https://www.talos.dev/latest/learn-more/process-capabilities/
+
+.. note::
+
+    The official Talos Linux documentation already covers many different Cilium deployment
+    options inside their `Deploying Cilium CNI guide`_. Thus, this guide will only focus
+    on the, from a Cilium perspective, the most recommended one:
```
Comment (ttarczynski) [inline Documentation/installation/k8s-install-talos-linux.rst]: Should this rather be:

> Thus, this guide will only focus on the most recommended deployment option, from a Cilium perspective:

File: Documentation/installation/k8s-install-talos-linux.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+.. _`Talos Linux Kubernetes configurations`: https://www.talos.dev/latest/reference/configuration/v1alpha1/config/#Config.cluster
+
+**Install Cilium**
+
+To run Cilium with `Kube-Proxy replacement<kubeproxy-free>` enabled, it's required
+to configure ``k8sServiceHost`` and ``k8sServicePort``, and point them to the
+Kubernetes API. Luckily, Talos Linux offers KubePrism_ which allow it to access the
+Kubernetes API in a convenient, internal way, which solely relies on host
+networking. This KubePrism_ endpoint can be accessed from every Talos Linux node on
+``localhost:7445``.
```
Comment (ttarczynski) [inline Documentation/installation/k8s-install-talos-linux.rst]: I couldn't grasp what the "internal way" means here.
After reading about [kubeprism](https://www.talos.dev/v1.6/kubernetes-guides/configuration/kubeprism/), I guess that's about not going through the external cluster endpoint, but I think it's worth clarifying.

**Review by ttarczynski — CHANGES_REQUESTED** [review]
LGTM, with two minor comments.

File: Documentation/installation/k8s-install-talos-linux.rst (hunk 1 above)
Comment (PhilipSchmid) [reply Documentation/installation/k8s-install-talos-linux.rst]: Thanks for the feedback. I addressed it in the newest commit.

File: Documentation/installation/k8s-install-talos-linux.rst (hunk 2 above)
Comment (PhilipSchmid) [reply Documentation/installation/k8s-install-talos-linux.rst]: Thanks for the feedback. I addressed it in the newest commit.

**Comment by PhilipSchmid** [comment]
cc @tommyp1ckles

**Review by frezbo — APPROVED** [review]
looks good to me from Talos side

File: Documentation/installation/k8s-install-talos-linux.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      my-cluster https://mycluster.local:6443 \
+      --config-patch @patch.yaml
+
+.. _`Talos Linux Kubernetes configurations`: https://www.talos.dev/latest/reference/configuration/v1alpha1/config/#Config.cluster
+
+**Install Cilium**
+
+To run Cilium with `Kube-Proxy replacement<kubeproxy-free>` enabled, it's required
+to configure ``k8sServiceHost`` and ``k8sServicePort``, and point them to the
+Kubernetes API. Luckily, Talos Linux provides KubePrism_ which allow it to access
```
Comment (ghost) [inline Documentation/installation/k8s-install-talos-linux.rst]:
```suggestion
Kubernetes API. Luckily, Talos Linux provides KubePrism_ which allows it to access
```

**Review by ghost — APPROVED** [review]
I left one nitpick comment, but looks good, thanks!

**Review by nickolaev — APPROVED** [review]

File: Documentation/installation/k8s-install-talos-linux.rst (hunk 3 above)
Comment (PhilipSchmid) [reply Documentation/installation/k8s-install-talos-linux.rst]: Thanks! Applied, squashed & rebased.

**Review by aspsk — APPROVED** [review]

**Comment by PhilipSchmid** [comment]
Thanks, everyone, for reviewing and approving! I just rebased to upstream `main`. AFAICT, it is now ready to merge.

**Review by ttarczynski — APPROVED** [review]
LGTM

**Review by tommyp1ckles — APPROVED** [review]

**Pull request merged**